### PR TITLE
Add Embedded Compound Extension

### PR DESCRIPTION
**Pull request by hayesgm**
This patch adds the ability to be embedded as a Compound Extension. To do so, we simply need to use a wrapper for our Web3 provider, `RPCWeb3Provider`, which communicates via iframe communication. Additionally, once we detect we're loading in an embedded state (via `?embedded` search param), we don't show the top-nav header bar. Additionally, we assume we can auto-connect to a Web3 account, since the connection logic is handled by the Compound III app, itself.

Finally, this patch includes the ability to vote on all proposals, even if they are not in voting states. This is just for testing, but should be removed before merging.

### CODE REVIEW

File: components/layout.js [hunk 1]
```diff
@@ -72,9 +76,12 @@ export default function Layout({ children }) {
       </Head>
 
       {/* Header */}
-      <div>
-        <Header />
-      </div>
+      { !embedded ?
```
Comment (hayesgm) [inline components/layout.js]: TODO: Decide if we should show any header in embedded mode.

File: helpers/index.js [hunk 2]
```diff
@@ -268,17 +268,17 @@ const canVote = async (address, proposalId) => {
...
-      currentBlock > proposal.startBlock &&
-      currentBlock < proposal.endBlock - 2025
-    ) ||
-    proposal.canceled
-  ) {
-    const error = new Error("proposal voting period is not active");
-    error.code = 400;
-    throw error;
-  }
+  // if (
```
Comment (hayesgm) [inline helpers/index.js]: TODO: Remove testing tweak.

File: package.json [hunk 3]
```diff
@@ -25,5 +26,11 @@
     "unstated-next": "^1.1.0",
     "web3": "1.7.3",
     "web3modal": "^1.9.2"
+  },
+  "resolutions": {
```
Comment (hayesgm) [inline package.json]: TODO: Consider removing these two minor patches.

File: pages/index.js [hunk 4]
```diff
@@ -167,8 +167,7 @@ export default function Home({ defaultProposals, defaultPages }) {
                       >
                         Info
                       </button>
-                      {proposal.states[proposal.states.length - 1].state ===
-                      "active" ? (
+                      {true ? (
```
Comment (hayesgm) [inline pages/index.js]: TODO: Remove testing tweak.

**Review by hayesgm — COMMENTED** [review]
Can view deployment here https://comp-vote-xi.vercel.app and integrated as an extension here https://app-v3-ipfs.infura-ipfs.io/ipfs/Qmb4hpS6hRQxF8KdRYtr2AcEBdFY8RxLpVtS2xCsYh3eYm/extensions#comp_vote

**Review by Comp-Vote — APPROVED** [review]
LGTM

**Pull request merged**